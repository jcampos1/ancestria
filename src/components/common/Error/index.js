import React from 'react';
import PropTypes from 'prop-types';

class Error extends React.PureComponent {

    render() {
        const { errors } = this.props;

        if (errors.length === 0)
            return null;

        return (
            <div className="alert alert-danger" role="alert">
                {
                    errors.map((error, index) => (
                        <label key={`error${index}`}>{error}</label>
                    ))
                }
            </div>
        )
    }
}

Error.propTypes = {
    errors: PropTypes.array.isRequired
}

export default Error;