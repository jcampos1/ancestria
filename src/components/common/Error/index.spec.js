import React from 'react';
import {shallow} from 'enzyme';
import Error from 'components/common/Error/index';

describe('Error component', () => {
	it('Should render Error component when errors is empty', () => {
		const errors = [];
		const wrapper = shallow(<Error errors={errors}/>);
		expect(wrapper).toMatchSnapshot();

		expect(wrapper.find("div.alert label")).toHaveLength(0);
	});

	it('Should render Error component when errors is not empty', () => {
		const errors = ["Unable to log in with provided credentials."];
		const wrapper = shallow(<Error errors={errors}/>);
		expect(wrapper).toMatchSnapshot();

		expect(wrapper.find("div.alert label")).toHaveLength(errors.length);
	});
});