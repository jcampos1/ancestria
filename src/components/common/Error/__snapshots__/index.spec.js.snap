// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error component Should render Error component when errors is empty 1`] = `""`;

exports[`Error component Should render Error component when errors is not empty 1`] = `
<div
  className="alert alert-danger"
  role="alert"
>
  <label
    key="error_login{index}"
  >
    Unable to log in with provided credentials.
  </label>
</div>
`;
